examples/azure_key/AzureKeyCredential.cs

```diff
@@ -0,0 +1,137 @@
+using Azure.Core;
+using Azure.Identity;
+using Azure.Security.KeyVault.Keys;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using System;
+using System.Buffers.Binary;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using AzureSignatureAlgorithm = Azure.Security.KeyVault.Keys.Cryptography.SignatureAlgorithm;
+
+namespace Tmds.Ssh.AzureKeyExample;
+
+sealed class AzureKeyCredential : PrivateKeyCredential
+{
+    public AzureKeyCredential(string vaultName, string keyName) :
+        base((c) => GetAzureKey(vaultName, keyName, c), $"Azure Key Vault: {vaultName}/{keyName}")
+    { }
+
+    public static async ValueTask<string> GetAzurePubKey(string vaultName, string keyName)
```

Comment:
I think this method and the related `GetRsaPubKey`/`GetEcdsaPubKey` are for working with the public side of the key and they are unrelated to the private key handling needed for the library. Can you move them into `Program.cs` or into a separate class.

Reply:
I'd like the sample to focus on how to use `PrivateKeyCredential` with `LoadKey `and move code not needed for that out of the derived class.

If there is some code you want to share, you could make the `PrivateKeyCredential` `internal`.

Reply:
Makes sense, I've moved the code to `Program.cs`. I didn't want to remove it entirely because there is no other way to extract the public key for the `authorized_keys` making it harder to use.